oml/inference/base.py

```diff
@@ -0,0 +1,43 @@
+from pathlib import Path
+from typing import List, Optional
+
+import torch
+from torch import Tensor, nn
+from torch.utils.data import DataLoader
+from tqdm.auto import tqdm
+
+from oml.datasets.list_dataset import ListDataset
+from oml.transforms.images.utils import TTransforms, get_im_reader_for_transforms
+from oml.utils.images.images import TImReader
+from oml.utils.misc_torch import get_device, temporary_setting_model_mode
+
+
+# todo: use lightning for half precision and DDP
+@torch.no_grad()
+def inference_on_images(
+    model: nn.Module,
+    paths: List[Path],
+    transform: TTransforms,
+    num_workers: int,
```

Comment:
`num_workers: int = 0` is also ok

Reply:
I prefer to specify it directly because it's easy to forget
And then you may need a lot of time to notice that inference takes too long